lib/transports/TCP.js

```diff
@@ -247,4 +247,102 @@ Transport.prototype.close = function () {
     this.connection.destroy();
     this.closed = true;
     this.emit('close');
+};
+
+
+function ClientTransport(config){
+    var self = this;
+    
+    self.host = config.host;
+    self.port = config.port;
+    self.heloFromClient = config.heloFromClient;
+    self.heloFromServer = config.heloFromServer;
+
+    self.closed = false;
+    self.connection = new net.Socket();
+
+    self.connection.on('data', function (data) {
+        self.emit('data', data);
+    });
+
+    self.connection.on('close', function () {
+        console.log('c -> rcvd close');
+        if (!self.closed) {
+            self.closed = true;
+            self.emit('close');
+        }
+    });
+}
+util.inherits(ClientTransport,ClientTransportPrototype);
```

Comment:
You sure this works, and not `ReferenceError: ClientTransportPrototype is not defined` ?